mantidproject/mantid
# Fix multi dataset fitting bugs

**Description of work.**
Always use the single-dataset fit if only 1 spectrum is fitted.
Remove the crystal field functions from gui dialogs.
Improved handling of workspace groups.

**To test:**
1. Try sequential and simultaneous fitting of a single spectrum. The result should be the same.
2. Check that there are no CrystalField* functions in the Add Function dialog.
3. Check that workspace groups that don't contain matrix workspaces don't show in the Add Workspace dialog.


Fixes #24954

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
